### PR TITLE
[DOCS] Update Result Format Topic

### PR DESCRIPTION
**Pull request by kwcanuck**
In [DOC-406](https://greatexpectations.atlassian.net/browse/DOC-406), @talagluck requested updates to the [Result format topic](https://deploy-preview-8666.docs.greatexpectations.io/docs/reference/expectations/result_format/) to include "...explicit, process-oriented guidance around how to configure for different workflows". This PR implements the changes and the changes recommended in [this Slack thread](https://greatexpectationslabs.slack.com/archives/C04SFTSNBGC/p1688149576891249).

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [X] _Appropriate_ tests and docs have been updated

[DOC-406]: https://greatexpectations.atlassian.net/browse/DOC-406?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docusaurus/docs/reference/expectations/result_format.md [hunk 1]
```diff
@@ -3,67 +3,74 @@ title: Result format
...
-```python name="tests/integration/docusaurus/reference/core_concepts/result_format/result_format_complete_example_set"
-```
+To apply `result_format` to an Expectation, you pass it into the Expectation. 
+
+1. Run `$ great_expectations suite new` to obtain a Validator object instance.
+
+2. Run the following code to pass `result_format` into the Expectation:
+
+    ```python name="tests/integration/docusaurus/reference/core_concepts/result_format/result_format_complete_example_set"
+    ```
```
Comment (austiezr) [inline docs/docusaurus/docs/reference/expectations/result_format.md]:
```suggestion
To apply `result_format` to an Expectation, you pass it into the Expectation configuration on your Validator:

    ```python name="tests/integration/docusaurus/reference/core_concepts/result_format/result_format_complete_example_set"
    ```
```

File: docs/docusaurus/docs/reference/expectations/result_format.md [hunk 2]
```diff
@@ -3,67 +3,74 @@ title: Result format
...
-
+## Expectations result format values and fields
 
-
-### All Expectations
 | Fields within `result`                | BOOLEAN_ONLY                       |BASIC           |SUMMARY         |COMPLETE        |
 ----------------------------------------|------------------------------------|----------------|----------------|-----------------
 |    details (dictionary)               | Defined on a per-Expectation basis |
-### Column map Expectations (e.g. `ColumnMapExpectation`, `ColumnPairMapExpectation`, `MulticolumnMapExpectation`)
+
```
Comment (austiezr) [inline docs/docusaurus/docs/reference/expectations/result_format.md]: For some reason I can't edit this section, but it should be removed.

File: docs/docusaurus/docs/reference/expectations/result_format.md [hunk 3]
```diff
@@ -3,67 +3,74 @@ title: Result format
...
+## Column map Expectations result format values and fields
```
Comment (austiezr) [inline docs/docusaurus/docs/reference/expectations/result_format.md]:
```suggestion
## Column Map Expectations result format values and fields
```

File: docs/docusaurus/docs/reference/expectations/result_format.md [hunk 4]
```diff
@@ -78,12 +85,14 @@ Regardless of how Result Format is configured, `unexpected_list` is never render
 |    unexpected_index_list              |no              |no              |no              |yes             |
 |    unexpected_index_query             |no              |no              |no              |yes             |
 |    unexpected_list                    |no              |no              |no              |yes             |
-### Column aggregate Expectations
+
+## Column aggregate Expectations result format values and fields
```
Comment (austiezr) [inline docs/docusaurus/docs/reference/expectations/result_format.md]:
```suggestion
## Column Aggregate Expectations result format values and fields
```

**Review by austiezr — CHANGES_REQUESTED** [review]

File: docs/docusaurus/docs/reference/expectations/result_format.md (hunk 2 above)
Comment (kwcanuck) [reply docs/docusaurus/docs/reference/expectations/result_format.md]: @austiezr I have removed the _Expectations result format values and fields_ section.

**Review by austiezr — APPROVED** [review]
🚀

File: docs/docusaurus/docs/reference/expectations/result_format.md [hunk 5]
```diff
@@ -235,21 +225,19 @@ For example:
...
 
 ### Behavior for `COMPLETE`
-A `result` is generated with all available justification for why an Expectation was met or not. The format is  
-intended for debugging pipelines or developing detailed regression tests.
 
-Great Expectations has standard behavior for support for describing the results of `column_map_expectation` and
-`ColumnAggregateExpectation` Expectations.
+A `result` is generated with all available justification for why an Expectation was successful or unsuccessful. The format is intended for debugging pipelines or developing detailed regression tests.
+
+Great Expectations has standard behavior for support for describing the results of `column_map_expectation` and `ColumnAggregateExpectation` Expectations.
```
Comment (Rachel-Reverie) [inline docs/docusaurus/docs/reference/expectations/result_format.md]: This reads a bit awkwardly with the double 'for' in it.  Maybe something like:
```suggestion
Great Expectations implements standard behaviors to support describing the results of `column_map_expectation` and `ColumnAggregateExpectation` Expectations.
```
...not a strong recommendation.  Just seemed the wording here could be better.

**Review by Rachel-Reverie — APPROVED** [review]
One minor comment where I thought we might be able to word things better, but no strong recommendation.  Approved so you won't be blocked if you decide to revise that portion (or not).

**Pull request merged**